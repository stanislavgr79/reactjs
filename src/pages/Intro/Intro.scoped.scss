.show_movie_popup {
    height: 100vh;
    overflow: hidden;
}