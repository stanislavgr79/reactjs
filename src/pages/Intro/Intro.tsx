import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { AppState } from '../../redux/redux-store';

import Header from '../../components/structure/Header';
import Footer from '../../components/structure/Footer';
import DetailMovieTop from '../../components/container/DetailMovieTop';

import { Redirect, useLocation, useParams } from 'react-router';
import { fetchGetMovieById } from '../../redux/actions/movies-actions';
import { updateLocation } from '../../redux/actions/search-actions';

const ContentIntro = React.lazy(() => import('../../components/structure/ContentIntro'));

import './Intro.scoped.scss';

interface Props {
  id: string | undefined;
}

export default function Intro(): JSX.Element {
  const dispatch = useDispatch();
  const { showPopup, dataStatus } = useSelector((store: AppState) => {
    return store.moviesStore;
  });
  const { id }: Props = useParams();
  const location = useLocation();

  useEffect(() => {
    if (id) {
      dispatch(updateLocation(location.pathname));
      dispatch(fetchGetMovieById(parseInt(id)));
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [dispatch, id]);

  let content;

  if (dataStatus == 'failed') {
    content = <Redirect to="/404" />;
  } else {
    content = (
      <div className={`intro_page ${!showPopup ? '' : 'show_movie_popup'}`}>
        {id && showPopup ? <DetailMovieTop /> : <Header />}
        <ContentIntro />
        <Footer />
      </div>
    );
  }

  return <>{content}</>;
}
