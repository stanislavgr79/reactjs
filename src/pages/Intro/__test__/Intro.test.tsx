import React from 'react';
import { render } from '@testing-library/react';
import ReactRouter from 'react-router';
import { BrowserRouter, MemoryRouter, Route } from 'react-router-dom';
import renderer from 'react-test-renderer';

import Intro from '../Intro';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

// const mockLocation = '/movies';
// beforeEach(() => {
//   jest.spyOn(location, 'useLocation').mockReturnValue(mockLocation);
// });

describe('Intro', () => {
  const initialState = { moviesStore: { showPopup: false, dataStatus: 'success' } };
  const mockStore = configureStore();
  let store;

  // jest.mock('react-router', () => ({
  //   // ...jest.requireActual('react-router'),
  //   useParams: () => ({
  //     id: undefined,
  //   }),
  //   useRouteMatch: () => ({ url: '/movies' }),
  // }));

  test('snapshot renders', () => {
    store = mockStore(initialState);
    jest.spyOn(ReactRouter, 'useParams').mockReturnValue({ id: undefined });
    
    // const tree = render(
    //   <Provider store={store}>
    //     <Intro />
    //   </Provider>,
    // );

    const tree = renderer
      .create(
        <BrowserRouter>
          <Provider store={store}>
            <Route path="/movies">
              <Intro />
            </Route>
          </Provider>
        </BrowserRouter>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
    // expect(tree).toContainHTML('<div class="header">');
  });




  test('snapshot renders2', () => {
    store = mockStore(initialState);
    jest.spyOn(ReactRouter, 'useParams').mockReturnValue({ id: '123' });
    // const tree = render(
    //   <Provider store={store}>
    //     <Intro />
    //   </Provider>,
    // );

    const tree = renderer
      .create(
        <BrowserRouter>
          <Provider store={store}>
            <Route path="/movies/123">
              <Intro />
            </Route>
          </Provider>
        </BrowserRouter>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
    // expect(tree).toContainHTML('<div class="header">');
  });
});
