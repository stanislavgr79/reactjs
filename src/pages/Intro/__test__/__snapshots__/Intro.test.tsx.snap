// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro snapshot renders 1`] = `null`;

exports[`Intro snapshot renders2 1`] = `null`;
